tests/unit/test_blob.py

```diff
@@ -2577,6 +2577,12 @@ def _do_upload_helper(
             if_metageneration_not_match,
             **timeout_kwarg
         )
+
+        # Adjust num_retries expectations to reflect the conditional default in
+        # _do_upload()
+        if num_retries is None and if_metageneration_match is None:
```

Comment:
Can you comment on to what extent the conditional is covered by test cases?

Reply:
@tritone the `cover` session under `nox ` requires (since the merge of #308) 100% coverage (branch and statements).

Reply:
Ah yeah that's right, good point.

Reply:
Branch coverage of the code I've added is 100% according to the coverage tool. We are exercising scenarios where if_metageneration_match = true and num_retries is not None.